File: python/get-ith-bit.py
def getIthBit(int, idx):
  return 1 if (int & (1 << idx)) != 0 else 0;

def setIthBit(int, idx):
  return (int | (1 << idx))

def clearIthBit(int, idx):
  return int & ~(1 << idx)

def clearRightmostBit(int):
  return int & (int - 1)

def getParity(int):
  parity = True
  while int > 0:
    int = int & (int - 1)
    parity = not parity
  return parity

def getHammingDistance(x, y):
  test = x ^ y
  count = 0
  while test is not 0:
    test = test & (test - 1)
    count += 1
  return count

File: get-ith-bit.js
const getIthBit = function(int, idx) {
  return ((int & (1 << idx)) !== 0) ? 1 : 0;
};

const setIthBit = function(int, idx) {
  return (int | (1 << idx));
};

const clearIthBit = function(int, idx) {
  return int & ~(1 << idx);
};

const clearRightmostBit = function(int) {
  return int & (int - 1);
};

const getParity = function(int) {
  let parity = true;
  while (int) {
    int = int & (int - 1);
    parity = !parity;
  }
  return parity;
};

const getHammingDistance = function(x, y) {
  let test = x ^ y;
  let count = 0;
  while (test) {
    test = test & (test - 1);
    count++;
  }
  return count;
}

console.log(getIthBit(8, 2))